orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationAiModel.java

```diff
@@ -99,26 +105,45 @@ public class OrchestrationAiModel {
   public static final OrchestrationAiModel NOVA_MICRO =
       new OrchestrationAiModel("amazon--nova-micro");
 
-  /** Azure OpenAI GPT-3.5 Turbo chat completions model */
+  /**
+   * Azure OpenAI GPT-3.5 Turbo model
+   *
+   * @deprecated This model is deprecated on AI Core with a planned retirement on 2025-02-13. The
+   *     suggested replacement model is {@link OrchestrationAiModel#GPT_4O_MINI}.
+   */
+  @Deprecated
   public static final OrchestrationAiModel GPT_35_TURBO = new OrchestrationAiModel("gpt-35-turbo");
 
   /**
-   * Azure OpenAI GPT-3.5 Turbo chat completions model
+   * Azure OpenAI GPT-3.5 Turbo model
    *
-   * @deprecated This model is not usable anymore. It is retired on AI Core since 2025-02-13.
+   * @deprecated This model is deprecated on AI Core with a planned retirement on 2025-02-13. The
+   *     suggested replacement model is {@link OrchestrationAiModel#GPT_4O_MINI}.
    */
   @Deprecated
   public static final OrchestrationAiModel GPT_35_TURBO_16K =
       new OrchestrationAiModel("gpt-35-turbo-16k");
 
-  /** Azure OpenAI GPT-3.5 Turbo chat completions model */
+  /**
+   * Azure OpenAI GPT-3.5 Turbo model
+   *
+   * @deprecated This model is deprecated on AI Core with a planned retirement on 2025-02-22. The
+   *     suggested replacement model is {@link OrchestrationAiModel#GPT_4O_MINI}.
+   */
+  @Deprecated
   public static final OrchestrationAiModel GPT_35_TURBO_0125 =
       new OrchestrationAiModel("gpt-35-turbo-0125");
 
-  /** Azure OpenAI GPT-4 chat completions model */
+  /** Azure OpenAI GPT-4 model */
   public static final OrchestrationAiModel GPT_4 = new OrchestrationAiModel("gpt-4");
 
-  /** Azure OpenAI GPT-4-32k chat completions model */
+  /**
+   * Azure OpenAI GPT-4 model
+   *
+   * @deprecated This model is deprecated on AI Core with a planned retirement on 2025-05-13. The
+   *     suggested replacement model is {@link OrchestrationAiModel#GPT_4O}.
```

Comment:
(Question)

Just to confirm, we want to expose this dynamic information (deprecation date and replacement), rather than suggesting SAP Note directly? @jjtang1985?

Reply:
This was initially proposed by Till.  
From consumer perspective, it's convenient to have this information as Java annotation.
If it does not cost too much time, I would like to have them.